modules/fedimint-ln-client/src/lib.rs

```diff
@@ -493,16 +493,23 @@ impl LightningClientModule {
             .await?;
         let absolute_timelock = consensus_height + OUTGOING_LN_CONTRACT_TIMELOCK;
 
-        let contract_amount = {
-            let invoice_amount_msat = invoice
-                .amount_milli_satoshis()
-                .ok_or(anyhow::anyhow!("MissingInvoiceAmount"))?;
-            // TODO: better define fee handling
-            // Add 1% fee margin
-            let contract_amount_msat = invoice_amount_msat + (invoice_amount_msat / 100);
-            Amount::from_msats(contract_amount_msat)
+        // Compute amount to lock in the outgoing contract
+        let invoice_amount_msat = invoice
+            .amount_milli_satoshis()
+            .ok_or(anyhow::anyhow!("MissingInvoiceAmount"))?;
+
+        let fees = gateway.fees;
+        let base_fee = fees.base_msat as u64;
+        let margin_fee: u64 = if fees.proportional_millionths > 0 {
+            let fee_percent = 1000000 / fees.proportional_millionths as u64;
```

Comment:
What does 1000000 represent? Would prefer constant

Reply:
huh. it's just the number 1 million used to compute percentage here :)